### PR TITLE
ENH: various improvements to summarize

### LINKED ISSUE #23: summarize is very slow
probably because of the ``scipy.optimize`` call...

### LINKED ISSUE #32: remove h1: Count per feature detail
This is unnecessary with the current layout.

### LINKED ISSUE #39: summarize fails with a traceback if passed a table with zero or one sample
This can happen after overzealous filtering. 

```bash
Traceback (most recent call last):
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/bin/qiime", line 11, in <module>
    load_entry_point('q2cli', 'console_scripts', 'qiime')()
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/lib/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/lib/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/lib/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/lib/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/lib/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/caporaso/Google Drive/code/qiime2/q2cli/q2cli/commands.py", line 210, in __call__
    results = action(**arguments)
  File "<decorator-gen-138>", line 2, in summarize
  File "/Users/caporaso/Google Drive/code/qiime2/qiime2/qiime/core/callable.py", line 221, in callable_wrapper
    output_types)
  File "/Users/caporaso/Google Drive/code/qiime2/qiime2/qiime/core/callable.py", line 418, in _callable_executor_
    ret_val = callable(output_dir=temp_dir, **view_args)
  File "/Users/caporaso/Google Drive/code/qiime2/q2-feature-table/q2_feature_table/_summarize.py", line 67, in summarize
    sample_counts_ax = sns.distplot(sample_counts, kde=False, rug=True)
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/lib/python3.5/site-packages/seaborn/distributions.py", line 209, in distplot
    bins = min(_freedman_diaconis_bins(a), 50)
  File "/Users/caporaso/miniconda3/envs/qiime2-dev/lib/python3.5/site-packages/seaborn/distributions.py", line 30, in _freedman_diaconis_bins
    h = 2 * iqr(a) / (len(a) ** (1 / 3))
TypeError: len() of unsized object
```

### LINKED ISSUE #40: replace all occurrences of "count" with "frequency"

### LINKED ISSUE #41: remove max_frequency_even_sampling_depth functionality from summarize
This is the latest failed attempt at guessing an even sampling depth. It is never what we want to use in practice (in my experience) so we should drop it.

### PR DESCRIPTION
**Pull request by gregcaporaso**
fixes #41
fixes #39
fixes #32
fixes #23
partially fixes #40 

Some example visualizations:
 * [empty table](https://view.qiime2.org/?f=https://dl.dropboxusercontent.com/u/2868868/temp/empty-table.qzv)
 * [one sample table](https://view.qiime2.org/?f=https://dl.dropboxusercontent.com/u/2868868/temp/one-sample-table.qzv)
 * [moving pictures tutorial table](https://view.qiime2.org/?f=https://dl.dropboxusercontent.com/u/2868868/temp/table.qzv)

### CODE REVIEW

**Comment by jairideout** [comment]
what's left to do for #40? was it more involved than search/replace?

File: q2_feature_table/_summarize/_visualizer.py [hunk 1]
```diff
@@ -47,42 +46,40 @@ def view_taxa_data(output_dir: str, data: pd.Series) -> None:
 
 
 def summarize(output_dir: str, table: biom.Table) -> None:
+    number_of_samples = len(table.ids(axis='sample'))
```
Comment (jairideout) [inline q2_feature_table/_summarize/_visualizer.py]: Can you use `table.shape` instead? It'll be more efficient

File: q2_feature_table/_summarize/_visualizer.py [hunk 2]
```diff
@@ -47,42 +46,40 @@ def view_taxa_data(output_dir: str, data: pd.Series) -> None:
...
+    number_of_features = len(table.ids(axis='observation'))
+
     sample_summary, sample_counts = _count_summary(table, axis='sample')
+    if number_of_samples > 1:
+        sample_counts_ax = sns.distplot(sample_counts, kde=False, rug=True)
+        sample_counts_ax.set_xlabel('Frequency of sample')
```
Comment (jairideout) [inline q2_feature_table/_summarize/_visualizer.py]: "Frequency of sample" is a little confusing, it seems to imply the number of times a sample was observed. Do you think "Frequency per sample" makes sense? That title would also match the table titles you're using.

Same question about "Frequency of feature" below.

**Comment by gregcaporaso** [comment]
> what's left to do for #40? was it more involved than search/replace?

To do this for the whole code base will involve some interface changes (e.g., ``qiime feature-table rarefy --p-counts-per-sample``, so I was going to do it in a separate PR. I'll do that, unless you really prefer that it be in this PR.

**Comment by jairideout** [comment]
Makes sense, let's leave it as-is in this PR.

**Comment by gregcaporaso** [comment]
@jairideout, ready for re-review. I addressed both of your comments, and the example artifacts have been updated.

**Comment by jairideout** [comment]
Thanks @gregcaporaso!

**Pull request merged**